src/ParseFile.js

```diff
@@ -348,33 +355,43 @@ const DefaultController = {
     return CoreManager.getRESTController().request('POST', path, data, options);
   },
 
-  download: function(uri) {
+  download: function(uri, options) {
     if (XHR) {
-      return this.downloadAjax(uri);
+      return this.downloadAjax(uri, options);
     } else if (process.env.PARSE_BUILD === 'node') {
       return new Promise((resolve, reject) => {
+        let aborted = false;
         const client = uri.indexOf('https') === 0
           ? require('https')
           : require('http');
-        client.get(uri, (resp) => {
+        const req = client.get(uri, (resp) => {
           resp.setEncoding('base64');
           let base64 = '';
           resp.on('data', (data) => base64 += data);
           resp.on('end', () => {
+            if (aborted) {
+              return resolve({});
```

Comment:
this is returning a resolved Promise to event distributor, does it actually do anything?  I doubt it.